src/routes/accounts.js

```diff
@@ -0,0 +1,114 @@
+import { list, readOne, deleteOne, deleteMany, patchOne, createOne } from 'mongoose-crudl'
+
+import allowAccessTo from 'bearer-jwt-auth'
+import AccountModel from '../models/Account.js'
+import UserModel from '../models/User.js'
+import Email from '../helpers/Email'
+import { ConflictError } from 'standard-api-errors'
+import fs from 'fs'
+import crypto from 'crypto'
+import handlebars from 'handlebars'
+import path from 'path'
+import jwt from 'jsonwebtoken'
+import { fileURLToPath } from 'url'
+const __dirname = path.dirname(fileURLToPath(import.meta.url))
+const registration = fs.readFileSync(path.join(__dirname, '..', 'email-templates', 'registration.html'), 'utf8')
+
+export default (apiServer) => {
+  const secrets = process.env.SECRETS.split(' ')
+
+  apiServer.get('/v1/accounts/check-availability', async req => {
+    let availability = false
+    const response = await list(AccountModel, { urlFriendlyName: req.query.urlFriendlyName })
+    if (response.result.count > 0) {
+      availability = true
+    }
+    return {
+      status: availability ? 200 : 404,
+      result: {
+        availability
+      }
+    }
+  })
+
+  apiServer.get('/v1/accounts/', async req => {
+    allowAccessTo(req, secrets, [{ type: 'admin' }])
+    const response = await list(AccountModel, req.params, req.query)
+    return response
+  })
+
+  apiServer.post('/v1/accounts/', async req => {
+    allowAccessTo(req, secrets, [{ type: 'admin' }])
+    const response = await createOne(AccountModel, req.params, req.body)
+    return response
+  })
+
+  apiServer.get('/v1/accounts/:id', async req => {
+    allowAccessTo(req, secrets, [{ type: 'admin' }, { type: 'user' }])
```

Comment:
The user should only be able to access this route if they are associated with the account. We should check the accounts.id on the token as well.